platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -752,9 +852,8 @@ public V remove(final K key) {
             final VirtualLeafRecord<K, V> sibling = records.findLeafRecord(lastLeafSibling, true);
             assert sibling != null;
             cache.clearLeafPath(lastLeafSibling);
-            cache.deleteInternal(lastLeafParent);
```

Comment:
Removing this line caused the `VirtualMapHashingTest.internalNodeSynchronization` test to fail.

Reply:
Also note that the deleted line used `lastLeafParent`, not `lastLeafSibling` as in the `clearLeafPath()` call above it. Deleting a hash in `clearLeafPath()` is not thus equivalent to the removed line, which needs to be restored.

Reply:
I replaced `cache.deleteInternal(lastLeafParent)` with `cache.deleteHash(lastLeafParent)`. It looks like it should be sufficient.